base/cmd.jl

```diff
@@ -242,6 +243,29 @@ setenv(cmd::Cmd, env::Pair{<:AbstractString}...; dir="") =
     setenv(cmd, env; dir=dir)
 setenv(cmd::Cmd; dir="") = Cmd(cmd; dir=dir)
 
+"""
+    addenv(command::Cmd, env...)
+
+Merge new environment mappings into the given `Cmd` object, returning a new `Cmd` object.
+Duplicate keys are replaced.
+"""
+function addenv(cmd::Cmd, env::Pair{<:AbstractString}...)
+    new_env = Dict{AbstractString,AbstractString}()
+    if cmd.env !== nothing
+        for (k, v) in split.(cmd.env, "=")
+            new_env[k] = v
+        end
+    end
+    for (k, v) in env
+        new_env[k] = v
+    end
+    return setenv(cmd, new_env)
+end
+
+function addenv(cmd::Cmd, env::Vector{<:AbstractString})
+    return addenv(cmd, [k => v for (k, v) in split.(env, "=")]...)
```

Comment:
With the suggestion above you don't need this splatting
```suggestion
    return addenv(cmd, Dict{String,String}(k => v for (k, v) in split.(env, "=")))
```